Add runner class name to pipeline hooks + introduce `after_command_run` CLI hook spec (#1309)
kedro-org/kedro: docs/source/extend_kedro/hooks.md

Lastly, Kedro defines a small set of CLI hooks that inject additional behaviour around execution of a Kedro CLI command:

* `before_command_run`
* `after_command_run`

> This is what the [`kedro-telemetry`](https://github.com/kedro-org/kedro-plugins/tree/main/kedro-telemetry) plugin relies on these under the hood in order to be able to collect CLI usage statistics.

Reviewer: ```suggestion
The [`kedro-telemetry`](https://github.com/kedro-org/kedro-plugins/tree/main/kedro-telemetry) plugin relies on these hooks under the hood in order to be able to collect CLI usage statistics.
```
